### DISCUSSION TITLE
[Questions] potential_evapotranspiration with FAO providing net radiation directly

### DISCUSSION THREAD

**Discussion by saschahofmann** [Ouranosinc/xclim#1853] [Questions]
### Setup Information

* Xclim version: 0.51.0

### Context

I am trying to compute `potential_evapotranspiration` with the `FAO_PM98` method but I don't have the 4 radiation parts separated instead I have ssr which I think is the net radiation already. I think I can hack my way around it by providing, rsds=0 and the rest as 0s but I was wondering whether it would make sense to either provide the 4 components **or** the net radiation directly or to extract the net_radiation computation and have a single argument for it to the pet function.

I have an additional caveat that I only have monthly climatologies for my radiation but I guess that's a completely different problem.

### Steps To Reproduce

```
$ pip install foo --bar
```

**Comment by coxipi** [comment]
I think it's reasonable, other PET formulas also use `ssr`. It will make the function even messier, but it's alright. I will open a PR. 

>  I only have monthly climatologies for my radiation

Yes, the FAO-PM98 is an empirical formula, in xclim it's currently hard coded for daily timeseries. I think it would only be a matter of computing a different 
```python
G = 0  # Daily soil heat flux density [MJ m-2 d-1]
```
the monthly one will depend on some variables.

**Reply by saschahofmann** [reply]
I now have the hourly era5 data for net solar radiation (ssr) and net thermal radiation (str). My plan is to provide them as rsus and rlus and the other two parts as 0 but I am not sure whether I should do sum or mean aggregation for the daily values?

**Comment by coxipi** [comment]
Hmm wait, `ssr` is Surface Solar Radiation? So it's the net shortwave radiation? Do you know a variable name for net radiation? 

EDIT: Ok, according to: https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html, 
`rss` is `surface_net_downward_shortwave_flux`

`ssr` is used in ERA5 for "surface_net_downward_shortwave_flux" as well.

**Reply by saschahofmann** [reply]
Yeah I am still trying to figure out whether its the same as the one computed inside the function but I don't think so.

**Reply by saschahofmann** [reply]
Or is it just `rsds - rsus `

**Reply by saschahofmann** [reply]
From the link you posted:
> The surface called "surface" means the lower boundary of the atmosphere. "shortwave" means shortwave radiation. "Downward" indicates a vector component which is positive when directed downward (negative upward). Net downward radiation is the difference between radiation from above (downwelling) and radiation from below (upwelling). In accordance with common usage in geophysical disciplines, "flux" implies per unit area, called "flux density" in physics.

It sounds like its missing the longwave radiation part?

**Reply by coxipi** [reply]
If you type `surface_net_downward_longwave_flux`  you will find `rls`.

**Reply by saschahofmann** [reply]
Yeah. I am just checking whether something like this is available in ERA5 but they only have have longwave (thermal) in the MARS catalogue 
<img width="458" alt="image" src="https://github.com/user-attachments/assets/c39a6a63-205b-48e2-ba2b-606e15d83db7">

**Reply by coxipi** [reply]
Monthly ERA5-Land has: `rsds, rlds, rss, rls`

![image](https://github.com/user-attachments/assets/813bc590-f213-4d20-8e2c-0d7728485fab)

https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-land-monthly-means?tab=form

**Reply by saschahofmann** [reply]
Yeah. I am also reading through the Allen 98 paper, linked in the docs and learning a lot. If I understand correctly, the solar flux arrives in short-waves hence only that is available, it will be transformed to thermal/long-wave. So I guess with these the net radiation should be Surface net short-wave (solar) radiation -  Surface net long-wave (thermal) radiation, assuming that the thermal radiation represents an energy loss (=upwards flux) ? 

CDS servers seem to be pretty busy right now but I am going to download some of these datasets and test my hypotheses.

**Reply by saschahofmann** [reply]
Another question, above you're suggesting to use monthly era5 land data in that case I would need to resample my other datasets to monthly as well right?

**Reply by coxipi** [reply]
Sorry, I didn't see your questions until now. 

1.  Short == solar; long == thermal.  They are used as synonyms.

> I guess with these the net radiation should be Surface net short-wave (solar) radiation - Surface net long-wave (thermal) radiation

2. No I don't think that's right. Net>0 if it points downwards et Net<0 if it point up. I don't think there's a difference between shortwave/longwave on that matter, but you should double check to be sure. 
 `Net radiation = Net shortwaves + Net longwaves = (down_shortwave - up_shortwave) + (down_longwave - up_longwave)`

>  above you're suggesting to use monthly era5 land

I think I was just working with your constraint: "I have an additional caveat that I only have monthly climatologies for my radiation."  so I was searching an example similar to yours. If you can find daily radiations that suit your needs, that's good. But, indeed, if you only find monthly radiations, you should work with variables that are all monthly in the PET formula.

**Comment by saschahofmann** [comment]
Stumbled over this repo https://github.com/woodcrafty/PyETo that has great implementations of the pet functions a little closer to the paper.
